samples/boards/st/power_mgmt/standby_shutdown/src/main.c

```diff
@@ -27,33 +28,24 @@
 #error "Unsupported board: sw0 devicetree alias is not defined"
 #endif
 
+#define WKUP_SRC_NODE DT_ALIAS(wkup_src)
+#if !DT_NODE_HAS_STATUS_OKAY(WKUP_SRC_NODE)
+#error "Unsupported board: wkup_src devicetree alias is not defined"
+#endif
+
+
 /* Semaphore used to control button pressed value */
 static struct k_sem button_sem;
 
 static int led_is_on;
 
-static const struct gpio_dt_spec button =
-	GPIO_DT_SPEC_GET_OR(SW0_NODE, gpios, {0});
+static const struct gpio_dt_spec button = GPIO_DT_SPEC_GET(WKUP_SRC_NODE, gpios);
 
 static const struct gpio_dt_spec led =
 	GPIO_DT_SPEC_GET(DT_ALIAS(led0), gpios);
 
 static struct gpio_callback button_cb_data;
 
-void config_wakeup_features(void)
-{
-	/* Configure wake-up features */
-	/* WKUP2(PC13) only , - active low, pull-up */
-	/* Set pull-ups for standby modes */
-	LL_PWR_EnableGPIOPullUp(LL_PWR_GPIO_C, LL_PWR_GPIO_BIT_13);
-	LL_PWR_IsWakeUpPinPolarityLow(LL_PWR_WAKEUP_PIN2);
```

Comment:
I don't think setting wake-up pin polarity to low is incorrect, since it's the push-button that determines it, & in this case it's `Active low`.

Reply:
this is done through the WakeUp pin config, isn't it

Reply:
Yes, but only when `wkup-pins-pol` property is present in pwr node, which was removed in board overlay.
The presence of this property result in the definition of the `PWR_STM32_WKUP_PINS_POLARITY` flag
https://github.com/zephyrproject-rtos/zephyr/blob/03959a20f7834ffe50a51df2b2e45ef87d1b610b/soc/st/stm32/common/stm32_wkup_pins.c#L35
which guards the polarity cfg code
https://github.com/zephyrproject-rtos/zephyr/blob/03959a20f7834ffe50a51df2b2e45ef87d1b610b/soc/st/stm32/common/stm32_wkup_pins.c#L196